Library/OcTextInputLib/OcTextInputLocalLib.c

```diff
@@ -0,0 +1,73 @@
+/** @file
+   OcTextInputLocalLib - SimpleTextInputEx Protocol Compatibility (Local Variant)
+
+   This is the "local" variant for use by OpenShell, which uses constructor-based
+   approach for automatic protocol installation.
+
+   Copyright (c) 2025, OpenCore Team. All rights reserved.
+   SPDX-License-Identifier: BSD-2-Clause-Patent
+ **/
+
+#include <Uefi.h>
+#include <Protocol/SimpleTextInEx.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/DebugLib.h>
+#include <Library/OcTextInputLib.h>
+
+#include "OcTextInputLibInternal.h"
+
+/**
+   Install SimpleTextInputEx compatibility protocol using standard registration.
+
+   This function uses the standard gBS->InstallProtocolInterface method for
+   protocol installation, making it suitable for use within OpenShell.
+
+   @retval EFI_SUCCESS          Protocol installed successfully or already present
+   @retval EFI_ALREADY_STARTED  Protocol already exists, no action taken
+   @retval Others               Installation failed
+ **/
+EFI_STATUS
+EFIAPI
+OcInstallSimpleTextInputExLocal (
+  VOID
+  )
+{
+  // Call the internal implementation with standard registration
+  return OcInstallSimpleTextInputExInternal (FALSE);
+}
+
+/**
+   Constructor for OcTextInputLocalLib.
+
+   Automatically installs SimpleTextInputEx compatibility protocol when
+   the library is loaded (constructor-based approach).
+
+   @param  ImageHandle   The firmware allocated handle for the EFI image.
+   @param  SystemTable   A pointer to the EFI System Table.
+
+   @retval EFI_SUCCESS   The constructor succeeded.
+ **/
+EFI_STATUS
+EFIAPI
+OcTextInputLocalLibConstructor (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  DEBUG ((DEBUG_INFO, "OcTextInputLocalLib: Constructor called\n"));
+
+  // Automatically install SimpleTextInputEx compatibility
```

Comment:
Here and elsewhere, house (OC and edk2) style for comments would be:

```
//
// Automatically install SimpleTextInputEx compatibility.
//
```

However also preferable - here and elsewhere - would be to omit comments which directly state what the code does, when what the code does can be very directly read from the code. (I'll point out one or two other examples.)